streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -1331,6 +1307,20 @@ private void validateRackAwarenessConfiguration() {
         });
     }
 
+    private static void validateProcessingConfiguration(final String processingModeConfig) {
+        if (processingModeConfig.equals(EXACTLY_ONCE)) {
+            throw new ConfigException(String.format("Configuration parameter `%s` was removed in the 4.0.0 release. " +
+                "Please use `%s` instead. Note that this requires broker version 2.5+ so you should prepare "
+                + "to upgrade your brokers if necessary.", EXACTLY_ONCE, EXACTLY_ONCE_V2));
```

Comment:
Also need to mention while changing the flag users should also follow the upgrade guide doing rolling bounces.

Reply:
Done